tests/unit/FirebaseObject.spec.js

```diff
@@ -322,6 +322,21 @@ describe('$FirebaseObject', function() {
       expect($scope.test).toEqual({$value: null, $id: obj.$id, $priority: obj.$priority});
     });
 
+    it('should delete $value if set to an object', function () {
```

Comment:
$value is a special case to represent primitives. On an object value, it would imply the associated `ref` had a child named `$value`, and according to the error code I'm seeing:
> Paths must be non-empty strings and can't contain ".", "#", "$", "[", or "]"

The same logic would apply if you set any other primitive: 
```javascript
ref.set('a string'); // -> {$id:'someId', $priority:null, $value:'a string'}
ref.set({text:'an object'}); // -> {$id:'someId', $priority:null, text:'an object'}
```
 previously this sequence would have resulted in:
```javascript
{$id:'someId', $priority:null, text:'an object', $value:'a string'}
```
The old value from when it was a primitive is left over. That can't be what we want.